src/Akeneo/Category/back/Application/Storage/ProcessCategoryUpdate.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Category\Application\Storage;
+
+use Akeneo\Category\Application\Storage\Update\CategoryUpdater;
+use Akeneo\Category\Application\Storage\Update\CategoryUpdaterRegistry;
+use Akeneo\Category\Application\Storage\Update\UpsertCategoryProperties;
+use Akeneo\Category\Application\Storage\Update\UpsertCategoryTranslations;
+use Akeneo\Category\Domain\Model\Category;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProcessCategoryUpdate
+{
+    // The order in which the updater will be executed
+    private array $updaterExecutionOrder = [
+        'properties' => null,
+        'translations' => null,
+    ];
+
+    public function __construct(
+        private CategoryUpdaterRegistry $categoryUpdaterRegistry
+    )
+    {
+    }
+
+    public function update(Category $categoryModel, array $userIntents):void
```

Comment:
Ton linter est bon ? 
```suggestion
    public function update(Category $categoryModel, array $userIntents): void
```